bug fix

janson13 opened the pull request:
fixed bug: changes lost when rename unsaved process

janson13 committed:
Update RenameProcessDefinitionDialog.java

janson13 committed:
Update WorkspaceOperations.java

janson13 committed:
Update plugin_ru.properties

janson13 committed:
Update WorkspaceOperations.java

janson13 committed:
Update WorkspaceOperations.java

janson13 committed:
Update RenameProcessDefinitionDialog.java

janson13 committed:
Create AlertDialog.java

Commit:
test1 commit

Commit:
test2 commit

dofs197 commented on plugins/ru.runa.gpd/src/ru/runa/gpd/util/WorkspaceOperations.java:
```diff
@@ -95,678 +98,719 @@
+				renameProcessDefinition(selection);
+				return;
+			} catch (Exception e) {
+				e.printStackTrace();
```
e.printStackTrace(); - плохая обработка ошибок

janson13 replied:
согласен, поторопился - сейчас подправлю.
обычно я так не делаю

dofs197 commented on plugins/ru.runa.gpd/src/ru/runa/gpd/util/WorkspaceOperations.java:
```diff
@@ -95,678 +98,719 @@
+			refreshResource(newDefinitionFolder);
+
+			openProcessDefinition(newDefinitionFile);			
+
```
зачем столько пустых строк?

dofs197 replied:
Просьба применить стиль форматирования из https://runawfe.org/rus/doc/DevelopersInfo#CodeStyle.

janson13 replied:
ок, учту. тоже поторопился

janson13 replied:
применил и закоммитил

dofs197 commented on plugins/ru.runa.gpd/src/ru/runa/gpd/ui/dialog/AlertDialog.java:
```diff
@@ -0,0 +1,50 @@
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Shell;
+
+public class AlertDialog extends Dialog {
```
Лишний код. В eclipse есть такие диалоги. Например ru.runa.gpd.ui.custom.Dialogs.confirm(String, String)

janson13 replied:
да, это лишний класс - закоммитил по ошибке, не знаю как отменить

dofs197 replied:
Просто удалить и закоммитить.

janson13 replied:
сделал

dofs197 replied:
Он отсюда исчезнуть должен, пока его вижу

janson13 replied:
разобрался - удалил

dofs197 commented on plugins/ru.runa.gpd/plugin_ru.properties:
```diff
@@ -272,6 +272,7 @@ button.change = \u0418\u0437\u043c\u0435\u043d\u0438\u0442\u044c
 button.edit = \u0420\u0435\u0434\u0430\u043a\u0442\u0438\u0440\u043e\u0432\u0430\u0442\u044c
 button.update = \u041e\u0431\u043d\u043e\u0432\u0438\u0442\u044c
 button.rename = \u041f\u0435\u0440\u0435\u0438\u043c\u0435\u043d\u043e\u0432\u0430\u0442\u044c
+alert.save_before_rename = Save before rename!
```
Это же для русского текста файл, в англ. - нет совсем, в итоге у меня выглядит так.
![Screenshot from 2020-06-02 15-34-08](https://user-images.githubusercontent.com/12091818/83520830-01a1c100-a4e7-11ea-9545-99bd827986b1.png)

janson13 replied:
добавил

dofs197 replied:
Я изменений здесь (в вебе) не вижу

janson13 replied:
добавил

dofs197 replied:
Всё равно не вижу. В plugin_ru.properties должны быть русские тексты, в plugin.properties - английские.

janson13 replied:
не знаю почему, но изменения в пропертях не коммитятся.
AlertDialog тоже удалил, но он остался в файлах на гитхабе, хотя 
тут пишет, что удален:
https://github.com/janson13/runawfe-free-devstudio/commit/78e80e8b8f9dddfdf37004cbfb6cf16b92e9b22d

janson13 replied:
всё, закоммитил

dofs197 commented on plugins/ru.runa.gpd/src/ru/runa/gpd/util/WorkspaceOperations.java:
```diff
@@ -95,678 +98,719 @@
+		dialog.open();
+	}
+
+	public static ProcessDefinition createNewProcessDefinition(IStructuredSelection selection,
```
Потестировал - не работает с композицией (см. SubprocessDefinition). alert-диалог спрашивает подтверждения, если да - то просто закрывает композицию, не предлагая её переименовать.

janson13 replied:
такой задачи не было поставлено. 
>см. SubprocessDefinition
- не понимаю где смотреть и как воспроизвести

janson13 replied:
всё, понял.  разобрались со Светланой

janson13 replied:
пофиксил, реквест отправил.

Commit:
fixed formatting,
added alert.save_before_rename to all properties files

Commit:
applyed runa code style

Commit:
fixed bug at rename subprocess definition

Commit:
fixed bug:  * in saved editors title
fixed old bug: editor closed after renaming (for process and subProcess)

dofs197 commented on plugins/ru.runa.gpd/src/ru/runa/gpd/ui/dialog/RenameProcessDefinitionDialog.java:
```diff
@@ -94,9 +97,13 @@ private void updateButtons() {
+        if (processSaved) {
+            shell.setText(Localization.getString("RenameProcessDefinitionDialog.title"));
+        } else {
+            shell.setText(" Process unsaved  ");
```
Нет локализации

Commit:
added alert to to localization
removed AlertDialog

Commit:
repeat commit

Commit:
fix properties

Commit:
fixed bug - changes in subprocesses not saved
            at renaming parent process.

fixed bug - double open editor panel from left side at
            first time opening.

fixed bug - parent editor closed at rename subprocess.

fixed bug - not select last open at renaming.

fixed bug?  process editor open at assinging
            process as composition.

Commit:
fixed bugs according:
https://rm.processtech.ru/issues/1379#15

Commit:
fixed 1379#19

Commit:
bugfix 1379 #27, #28

dofs197 commented:
Fixed in https://github.com/processtech/runawfe-free-devstudio/commit/3d524ea7a68625dd27fa45aab50e39164e13a68b
